### PR TITLE
Added redux container pattern to To-Dos list project

### PR DESCRIPTION
**Pull request by Abdi-MS**

### CODE REVIEW

File: src/components/OneToDo/OneToDo.jsx [hunk 1]
```diff
@@ -5,35 +5,27 @@ import SaveIcon from "@mui/icons-material/Save";
...
 
-  const dispatch = useDispatch();
-
-  const reduxToDoList = useSelector((state) => {
-    return state.toDo.toDoList;
-  });
-
   const editField = useRef();
 
   const editToDo = () => {
```
Comment (zain-sattar) [inline src/components/OneToDo/OneToDo.jsx]: editTodoHandler

File: src/components/ToDos/ToDos.jsx [hunk 2]
```diff
@@ -2,17 +2,13 @@ import React, { useEffect } from "react";
...
+import { useSelector, useDispatch, connect } from "react-redux";
 import { startedApp } from "../../redux/toDoSlice/toDoSlice";
 
-function ToDos() {
-  const ToDoList = useSelector((state) => {
-    return state.toDo.toDoList;
-  });
-
+function ToDos({ ToDoList, startApp }) {
   const dispatch = useDispatch();
```
Comment (zain-sattar) [inline src/components/ToDos/ToDos.jsx]: remove this line

File: src/redux/toDoSlice/toDoSlice.js [hunk 3]
```diff
@@ -14,6 +14,7 @@ const toDoSlice = createSlice({
       state.toDoList.push(action.payload);
     },
     editToDo: (state, action) => {
+      console.log(action.payload)
```
Comment (zain-sattar) [inline src/redux/toDoSlice/toDoSlice.js]: remove this console

Comment (Abdi-MS) [reply src/redux/toDoSlice/toDoSlice.js]: done

File: src/components/ToDos/ToDos.jsx (hunk 2 above)
Comment (Abdi-MS) [reply src/components/ToDos/ToDos.jsx]: done

File: src/components/OneToDo/OneToDo.jsx (hunk 1 above)
Comment (Abdi-MS) [reply src/components/OneToDo/OneToDo.jsx]: done

File: src/components/OneToDo/OneToDo.jsx [hunk 4]
```diff
@@ -5,35 +5,27 @@ import SaveIcon from "@mui/icons-material/Save";
...
+  const editToDoHandler = () => {
+    editToDo({
+      text: editField.current.value,
+      id: toDoList[index].id,
+    });
   };
 
   const deleteToDo = () => {
-    dispatch(reduxDeleteToDo({ delIndex: index, id: reduxToDoList[index].id }));
+    actionDeleteToDo({ delIndex: index, id: toDoList[index].id });
```
Comment (zain-sattar) [inline src/components/OneToDo/OneToDo.jsx]: actionDeleteToDo rename this to deleteTodo

File: src/components/OneToDo/OneToDo.jsx [hunk 5]
```diff
@@ -5,35 +5,27 @@ import SaveIcon from "@mui/icons-material/Save";
...
-      })
-    );
+  const editToDoHandler = () => {
+    editToDo({
+      text: editField.current.value,
+      id: toDoList[index].id,
+    });
   };
 
   const deleteToDo = () => {
```
Comment (zain-sattar) [inline src/components/OneToDo/OneToDo.jsx]: rename this to deleteTodo handler

File: src/components/OneToDo/OneToDo.jsx [hunk 6]
```diff
@@ -117,4 +107,19 @@ function OneToDo({ index }) {
   );
 }
 
-export default OneToDo;
+const mapStateToProps = (state) => {
```
Comment (zain-sattar) [inline src/components/OneToDo/OneToDo.jsx]: move this into a seperate file as container

Comment (zain-sattar) [reply src/components/OneToDo/OneToDo.jsx]: this one is still unresolved

**Review by zain-sattar — CHANGES_REQUESTED** [review]

File: src/components/OneToDo/OneToDo.jsx [hunk 7]
```diff
@@ -5,35 +5,23 @@ import SaveIcon from "@mui/icons-material/Save";
 import React, { useRef } from "react";
 import { useState } from "react";
 import "./OneToDo.css";
-import { useDispatch, useSelector } from "react-redux";
-import {
-  editToDo as reduxEditToDo,
-  deleteToDo as reduxDeleteToDo,
-} from "../../redux/toDoSlice/toDoSlice";
+import connectOneToDo from "../containers/OneToDoContainer";
```
Comment (zain-sattar) [inline src/components/OneToDo/OneToDo.jsx]: why we are still importing connectOneTodo here?

File: src/components/NewToDo/NewToDo.jsx [hunk 8]
```diff
@@ -1,21 +1,19 @@
 import React, { useState } from "react";
 import { TextField } from "@mui/material";
 import "./NewToDo.css";
-import { useDispatch } from "react-redux";
-import { addToDo as reduxAddToDo } from "../../redux/toDoSlice/toDoSlice";
-import {v4 as uuidv4} from 'uuid'
+import { v4 as uuidv4 } from "uuid";
+import connectNewToDo from "../containers/NewToDoContainer";
```
Comment (zain-sattar) [inline src/components/NewToDo/NewToDo.jsx]: import connectNewToDo from "../containers/NewToDoContainer";
no longer need to import this

File: src/components/OneToDo/OneToDo.jsx (hunk 7 above)
Comment (Abdi-MS) [reply src/components/OneToDo/OneToDo.jsx]: it's a default export so I didn't change the naming, just updated the import location to the container file

File: src/components/NewToDo/NewToDo.jsx (hunk 8 above)
Comment (Abdi-MS) [reply src/components/NewToDo/NewToDo.jsx]: deleted this line

**Review by zain-sattar — APPROVED** [review]

**Pull request merged**